apps/studio/src/server/modules/folder/folder.router.ts

```diff
@@ -65,28 +67,47 @@ export const folderRouter = router({
           }
         }
 
-        const folder = await db
-          .insertInto("Resource")
-          .values({
-            permalink,
-            siteId,
-            type: ResourceType.Folder,
-            title: folderTitle,
-            parentId: parentFolderId ? String(parentFolderId) : null,
-            state: ResourceState.Published,
-          })
-          .returning(["id"])
-          .executeTakeFirstOrThrow()
-          .catch((err) => {
-            if (get(err, "code") === PG_ERROR_CODES.uniqueViolation) {
-              throw new TRPCError({
-                code: "CONFLICT",
-                message: "A resource with the same permalink already exists",
-              })
-            }
-            throw err
+        const folder = await db.transaction().execute(async (tx) => {
+          const user = await tx
+            .selectFrom("User")
+            .where("id", "=", ctx.user.id)
+            .selectAll()
+            .executeTakeFirstOrThrow(() => new TRPCError({ code: "NOT_FOUND" }))
```

Comment:
thoughts: perhaps move this out of the transaction since it's read-only and that the transaction won't be too blocking?

Reply:
No strong opinions on this one, initially thought would be good to keep all in one transaction to maintain data integrity due to interleaving but I think extremely unlikely. Updated in 9fff24c507dd19fca1f26ccd2fba956c074169d8.